### ISSUE TITLE
Automatic insertion of related entities

### ISSUE THREAD

**Issue by carlosafonso** [carlosafonso/laravel-crud#8]
If there are two models, `Foo` and `Bar`, related so that `Bar` belongs to `Foo`, if the following input is received:

```
POST /foos
```

``` json
{
    "attribute_1": "abc",
    "attribute_2": "def",
    "bar": {
        "another_attribute": "ghi"
    }
}
```

Then the controller should automatically create the related entity `Bar` and associate it to the newly created `Foo`.

**Comment by carlosafonso** [comment]
We could check whether any of the input attributes matches an Eloquent relation in the associated model. If so, we should be able to create and attach the related entity.

We could even check what type of relation it is (belongsTo, hasMany...) and proceed as appropriate.

How about validation though? @namelivia

**Comment by namelivia** [comment]
Yeah, that's a good way to proceed. Also the child entities should be validated, probably getting the validation rules from their models.

**Comment by carlosafonso** [comment]
Alright. I'm worried about the hit on performance due to the dynamic relationship lookup. Maybe we could use a more static approach, but this is more annoying for the developer:

``` php
/* just came up with this, don't take it as something serious */
public function getRelatedEntities()
{
    return ['foo' => Foo::class];
}
```

**Comment by namelivia** [comment]
I can't ensure that would be neccesary due to performance issues, maybe we should not worry that much.

**Comment by namelivia** [comment]
My approach is to get the class methods by using a ReflecionClass, however this can be used to determine if the method exists or not. Is this approach the one you were thinkin on @carlosafonso ?

**Comment by carlosafonso** [comment]
Yes. Either that or [something like method_exists()](http://php.net/manual/en/function.method-exists.php)

**Comment by namelivia** [comment]
I've been also thinking that way, the thing is guessing the type of relationship.

**Comment by carlosafonso** [comment]
That's a good one. Maybe we could rely on the PHPDoc annotations (`@return`).

Or maybe this is just way too overkill and we're better off with a custom method that returns the accepted related entities. Maybe this is the easiest approach for now.

**Comment by carlosafonso** [comment]
And developers don't have to do anything if they don't want to use this feature.
